griptape/tasks/base_image_generation_task.py

```diff
@@ -72,27 +75,21 @@ def all_negative_rulesets(self) -> list[Ruleset]:
 
         return task_rulesets
 
-    def run(self) -> ImageArtifact:
-        image_artifact = self.image_generation_engine.generate_image(
-            prompts=[self.input.to_text()], rulesets=self.all_rulesets, negative_rulesets=self.negative_rulesets
-        )
-
-        if self.output_dir or self.output_file:
-            self._write_to_file(image_artifact)
-
-        return image_artifact
+    def _read_from_file(self, path: str) -> ImageArtifact:
+        self.structure.logger.info(f"Reading image from {os.path.abspath(path)}")
+        return ImageLoader().load(path)
 
     def _write_to_file(self, image_artifact: ImageArtifact) -> None:
         # Save image to file. This is a temporary workaround until we update Task and Meta
         # Memory to persist artifacts from tasks.
-        if self.output_file:
+        if self.output_file is not None:
             outfile = self.output_file
         else:
-            outfile = path.join(self.output_dir, image_artifact.name)
+            outfile = os.path.join(self.output_dir, image_artifact.name)  # type: ignore
 
-        if path.dirname(outfile):
-            os.makedirs(path.dirname(outfile), exist_ok=True)
+        if os.path.dirname(outfile):  # type: ignore
+            os.makedirs(os.path.dirname(outfile), exist_ok=True)  # type: ignore
 
-        with open(outfile, "wb") as f:
-            self.structure.logger.info(f"Saving [{image_artifact.to_text()}] to {os.path.abspath(outfile)}")
+        with open(outfile, "wb") as f:  # type: ignore
+            self.structure.logger.info(f"Saving [{image_artifact.to_text()}] to {os.path.abspath(outfile)}")  # type: ignore
```

Comment:
Why the ignores?

Reply:
Pylance strict typing error remediation; removed